raiden-network/light-client
# Monitors MS contract and emit event when it acts

SDK part of #1374 

**Short description**
Monitors MS contract and fires a `ms/balanceProof/sent` event through `Raiden.events$` observable if a Monitoring Service acted while we were offline and updated one of our channels.
This event's payload contain the following fields:
- `tokenNetwork`: token network from which the channel belongs
- `partner`: partner with which we had this channel
- `id`: channel Id
- `reward`: reward collected by MS from our UDC deposit for this service
- `nonce`: order of the state change on partner's channel state end which got updated
- `monitoringService`: address of the MS which acted and collected the reward
- `txHash`: hash of the transaction on-chain
- `txBlock`: when the transaction took place
- `confirmed`: if the transaction already got confirmed


**Definition of Done**


**Steps to manually test the change (dApp)**

Not that easy to test, but this worked here:
1. Ensure receiving is enabled for any channel value: e.g. `rateToSvt: { "<token_addr>": MaxUint256 }`
2. Ensure there's enough UDC deposit to pay for MS, e.g. 10SVT
3. While online, receive a transfer, check `RequestMonitoring` message was sent
4. Go offline
5. On partner (sender)'s side, close the channel
6. Don't come back online with receiver and wait `settleTimeout= 500` (default) blocks to pass
7. Channel becomes settleable, MS should have sent an update tx
8. Come back online with receiving node, see `ms/balanceProof/sent` action be detected and emitted
9. Either sender or receiver can then settle the channels, receiver should receive its due tokens
